extensions/resteasy-reactive/rest-client-microprofile/deployment/src/main/java/io/quarkus/resteasy/reactive/client/microprofile/deployment/ReactiveResteasyMpClientProcessor.java

```diff
@@ -0,0 +1,304 @@
+    void setupAdditionalBeans(
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            RestClientRecorder restClientRecorder) {
+        restClientRecorder.setRestClientBuilderResolver();
+        additionalBeans.produce(new AdditionalBeanBuildItem(RestClient.class));
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(HeaderContainer.class));
+    }
+
+    @BuildStep
+    void setupRequestCollectingFilter(BuildProducer<ContainerRequestFilterBuildItem> filters) {
+        filters.produce(new ContainerRequestFilterBuildItem(HeaderCapturingServerFilter.class.getName()));
+    }
+
+    @BuildStep
+    void addMpClientEnricher(BuildProducer<JaxrsClientEnricherBuildItem> enrichers) {
+        enrichers.produce(new JaxrsClientEnricherBuildItem(new MicroProfileRestClientEnricher()));
+    }
+
+    private void searchForJaxRsMethods(List<MethodInfo> listOfKnownMethods, ClassInfo startingInterface, CompositeIndex index) {
+        for (MethodInfo method : startingInterface.methods()) {
+            if (isRestMethod(method)) {
+                listOfKnownMethods.add(method);
+            }
+        }
+        List<DotName> otherImplementedInterfaces = startingInterface.interfaceNames();
+        for (DotName otherInterface : otherImplementedInterfaces) {
+            ClassInfo superInterface = index.getClassByName(otherInterface);
+            if (superInterface != null)
+                searchForJaxRsMethods(listOfKnownMethods, superInterface, index);
+        }
+    }
+
+    @BuildStep
+    void registerHeaderFactoryBeans(CombinedIndexBuildItem index,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        Collection<AnnotationInstance> annotations = index.getIndex().getAnnotations(DotNames.REGISTER_CLIENT_HEADERS);
+
+        for (AnnotationInstance registerClientHeaders : annotations) {
+            AnnotationValue value = registerClientHeaders.value();
+            if (value != null) {
+                Type clientHeaderFactoryType = value.asClass();
+                String factoryTypeName = clientHeaderFactoryType.name().toString();
+                if (!MicroProfileRestClientEnricher.DEFAULT_HEADERS_FACTORY.equals(factoryTypeName)) {
+                    additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(factoryTypeName));
+                }
+            }
+        }
+    }
+
+    @BuildStep
+    void addRestClientBeans(Capabilities capabilities,
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeans) {
+
+        CompositeIndex index = CompositeIndex.create(combinedIndexBuildItem.getIndex());
+        Set<AnnotationInstance> registerRestClientAnnos = new HashSet<>(index.getAnnotations(REGISTER_REST_CLIENT));
+        for (AnnotationInstance registerRestClient : registerRestClientAnnos) {
+            ClassInfo jaxrsInterface = registerRestClient.target().asClass();
```

Comment:
Do we actually need to generate a class here? Can't we just use a synthetic bean?

Reply:
Rest Client stubs that are injected through CDI have to support interceptors. Synthetic beans don't.